launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -193,6 +193,12 @@ List<String> buildClassPath(String appClassPath) throws IOException {
     boolean isTestingSql = "1".equals(getenv("SPARK_SQL_TESTING"));
     String jarsDir = findJarsDir(getSparkHome(), getScalaVersion(), !isTesting && !isTestingSql);
     if (jarsDir != null) {
+      // Place slf4j-api-* jar first to be robust
```

Comment:
For my own understanding, is this the test specific issue, or actual bug?

Reply:
Ya, it seems to be a real robustness issue. It randomly happens based on the class loading sequence.